lib/rules/disallow-newline-before-block-statements.js

```diff
@@ -90,15 +183,57 @@ module.exports.prototype = {
     },
 
     check: function(file, errors) {
-        file.iterateNodesByType('BlockStatement', function(node) {
-            var openingBrace = file.getFirstNodeToken(node);
-            var prevToken = file.getPrevToken(openingBrace);
+        var setting = this._setting;
+        if (setting === true) {
+            file.iterateNodesByType('BlockStatement', function(node) {
```

Comment:
still can consolidate this into one iteration over BlockStatement, no reason to split the two.